src/components/list/list.tsx

```diff
@@ -338,56 +339,56 @@ export class List implements InteractiveComponent, LoadableComponent {
     }
   };
 
+  private filterElements({
+    el,
+    filteredItems,
+    visibleParents,
+    visible
+  }: {
+    el: HTMLCalciteListItemElement | HTMLCalciteListItemGroupElement;
+    filteredItems: HTMLCalciteListItemElement[];
+    visibleParents: WeakSet<HTMLElement>;
+    visible?: boolean;
+  }): void {
+    const isVisible = filteredItems.includes(el as HTMLCalciteListItemElement) || visible;
+
+    el.hidden = visibleParents.has(el) ? false : !isVisible;
```

Comment:
I think this could be simplified as `el.hidden = !visibleParents.has(el) && !isVisible;`.